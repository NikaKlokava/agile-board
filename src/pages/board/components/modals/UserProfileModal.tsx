import { signOut } from "firebase/auth";
import { memo, useCallback } from "react";
import { useNavigate } from "react-router-dom";
import { auth } from "../../../../firebase";
import { useAppDispatch, useAppSelector } from "../../../../redux/hooks/hook";
import { resetBoards } from "../../../../redux/reducers/boardsSlice";
import { resetTasks } from "../../../../redux/reducers/tasksSlice";
import { Button } from "../../../../shared/components/button";
import { CloseIcon } from "../../../../shared/components/close_icon/CloseIcon";
import { ModalWrapper } from "../../../../shared/components/modal_wrapper";
import { StatisticsItem } from "../../../../shared/components/statistics_item";
import cl from "./modal_styles.module.css";

type Props = {
  onClose: () => void;
};

export const UserProfileModal = memo(({ onClose }: Props) => {
  const dispatch = useAppDispatch();
  const navigate = useNavigate();

  const tasks = useAppSelector((state) => state.tasks.tasks);
  const boards = useAppSelector((state) => state.boards.boards);

  const userBoards = boards.filter(
    (board) => board.usersEmail[0] === auth.currentUser?.email
  );

  const sharedBoards = boards.filter(
    (board) => board.usersEmail[0] !== auth.currentUser?.email
  );

  const handleSignOutClick = useCallback(() => {
    dispatch(resetTasks());
    dispatch(resetBoards());
    signOut(auth);
    navigate("/agile-board");
  }, [dispatch, navigate]);

  return (
    <ModalWrapper onWrapperClick={onClose}>
      <CloseIcon onClose={onClose} />
      <h2 className={cl.modal_title}>Profile</h2>
      <div className={cl.profile_avatar} />
      <div className={cl.container}>
        <p className={cl.title}>Email</p>
        <div className={cl.line} />
        <p>{auth.currentUser?.email}</p>
      </div>
      <div className={cl.container}>
        <p className={cl.title}>Statistics</p>
        <div className={cl.line} />
        <StatisticsItem name={"My boards"} item={userBoards?.length} />
        <StatisticsItem name={"Shared boards"} item={sharedBoards?.length} />
        <StatisticsItem name={"Tasks"} item={tasks?.length} />
      </div>
      <Button
        text={"Sign Out"}
        withIcon={false}
        testid={"sign_out_btn"}
        type={"button"}
        onClick={handleSignOutClick}
      />
    </ModalWrapper>
  );
});
