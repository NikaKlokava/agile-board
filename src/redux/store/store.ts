import { legacy_createStore } from "redux";
import { boardsReducer } from "../reducers/boardReducer";

const store = legacy_createStore(boardsReducer);

export default store;

// export type RootState = ReturnType<typeof store.getState>

// export interface RootState {
//     user: UserState;
//     loader: LoaderState;
//   }
// export type AppDispatch = typeof store.dispatch
