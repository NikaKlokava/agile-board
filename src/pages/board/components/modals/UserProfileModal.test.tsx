import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import store from "../../../../redux/store/store";
import { UserProfileModal } from "./UserProfileModal";

describe("Test the UserProfileModal component", () => {
  test("The UserProfileModal renders correctly", () => {
    const userProfileModalSnap = renderer
      .create(
        <Provider store={store}>
          <UserProfileModal onClose={() => console.log("close")} />
        </Provider>
      )
      .toJSON();
    expect(userProfileModalSnap).toMatchSnapshot();
  });
});
