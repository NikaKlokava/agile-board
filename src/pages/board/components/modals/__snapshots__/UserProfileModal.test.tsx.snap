// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the UserProfileModal component The UserProfileModal renders correctly 1`] = `
Array [
  <div
    className="modal_wrapper"
    onClick={[Function]}
  />,
  <div
    className="modal_container"
  >
     
    <h2
      className="modal_title"
    >
      Profile
    </h2>
    <div
      className="profile_avatar"
      style={
        Object {
          "background": "url(\\"undefined\\")",
        }
      }
    />
    <div
      className="container"
    >
      <p
        className="title"
      >
        Email
      </p>
      <div
        className="line"
      />
      <p />
    </div>
    <div
      className="container"
    >
      <p
        className="title"
      >
        Statistics
      </p>
      <div
        className="line"
      />
      <div
        className="statistics_item"
      >
        <p>
          My boards
        </p>
        <p>
          0
        </p>
      </div>
      <div
        className="statistics_item"
      >
        <p>
          Shared boards
        </p>
        <p>
          0
        </p>
      </div>
      <div
        className="statistics_item"
      >
        <p>
          Tasks
        </p>
        <p>
          0
        </p>
      </div>
    </div>
    <button
      className="button"
      data-testid="sign_out_btn"
      onClick={[Function]}
      type="button"
    >
      <div
        className="button_text"
      >
        Sign Out
      </div>
    </button>
  </div>,
]
`;
